pkg/api/v1alpha1/cluster_defaults.go

```diff
@@ -50,6 +50,11 @@ func setWorkerNodeGroupDefaults(cluster *Cluster) error {
 		logger.V(1).Info("First worker node group name not specified. Defaulting name to md-0.")
 		cluster.Spec.WorkerNodeGroupConfigurations[0].Name = "md-0"
 	}
+	for i := range cluster.Spec.WorkerNodeGroupConfigurations {
```

Comment:
```suggestion
	for _, wng := range cluster.Spec.WorkerNodeGroupConfigurations {
	    if wng.Count <= 0 {
	      wng.Count = 1
	    }
	
	}
	
```
```suggestion
	for i := range cluster.Spec.WorkerNodeGroupConfigurations {
```

Reply:
I tried the first suggestion initially, but it doesn't actually update the count in the cluster spec. That's why I used the index